hl/src/H5LTanalyze.l

```diff
@@ -123,6 +124,25 @@ OPQ_TAG          {return token(OPQ_TAG_TOKEN);}
 "\n"             { return 0; }
 
 %%
+
+/* Allocate a copy of `quoted` with the double quote character at
+ * the beginning and the one at the end both removed.  The caller is
+ * responsible for free()ing the copy.
+ */
+static char *
+trim_quotes(const char *quoted)
+{
+    size_t len = strlen(quoted);
+    char *trimmed;
+
+    assert(quoted[0] == '"' && quoted[len - 1] == '"');
+
+    trimmed = strdup(quoted + 1);
```

Comment:
Even MSVC 2015 supports enough C99 to do the declaration & initialization at once, as in: `char *trimmed = strdup(quoted + 1);`  Is there a reason you do this C89 style? (I've noticed most of HDF5 is the same.)